src/box/tuple_bloom.c

```diff
@@ -217,7 +217,18 @@ tuple_bloom_maybe_has(const struct tuple_bloom *bloom, struct tuple *tuple,
 	uint32_t h = HASH_SEED;
 	uint32_t carry = 0;
 	uint32_t total_size = 0;
-
+	if (bloom->version == TUPLE_BLOOM_VERSION_V2) {
```

Comment:
The function looks weird with the two loops. IMO it'd be better to fold the version check in the loop body.

Reply:
That would mean the check happens multiple times instead of one. I didn't measure performance of this place, but that surely would be suboptimal. Should we keep it like this then? Or do you still think that better check it on each iteration?